### DISCUSSION TITLE
Is it possible set proxy for new AI feature?

### DISCUSSION THREAD

**Discussion by wss29** [eclipse-theia/theia#14316] [Q&A]
Hi,
   the new ai feature in version 1.54.0 is very wonderful, we want to use it. but there is a problem with open ai
   1. I can not access open ai directly, so I used a proxy in my computer and tried a demo in the terminal of ubuntu, it works
as you can see in the following image, I asked open ai "Write a haiku that explains the concept of recursion", it answer me with "Endless paths unfold,  \neach step calls back to before—  \nself within itself."
![image](https://github.com/user-attachments/assets/834100f2-396e-4545-87bf-5011504e92ff)
   2. but when I set my OPENAI_API_KEY in theia setting I can not access, it gets a connection error. So I guess it may be necessary to set a proxy in theia for access open ai model
![截图 2024-10-15 14-44-41](https://github.com/user-attachments/assets/5d122253-8dd1-4c78-91f0-15e56bbab34e)
![image](https://github.com/user-attachments/assets/2a4da541-29c3-4466-bb34-17c2a8630c11)
![image](https://github.com/user-attachments/assets/268f21b1-9152-46ac-a961-61082958e28b)
  3. I am not sure it is the problem of proxy. does anyone has idea?

**Comment by sdirix** [comment]
Hi @wss29,

If you have a dedicated proxy URL for Open AI then you should be able to workaround the issue using the `Custom Open Ai Models` preference. In there you could for example add:
```
{
  "model": "gpt-4o",
  "id": "gpt-4o-proxy",
  "url": "{YOUR_PROXY_URL}"
}
```

The [official Open AI API also has proxy support](https://github.com/openai/openai-node/blob/3c326626c1035d43ee184cedc9084a7d399331df/README.md?plain=1#L584-L604), so I think it would make sense for us to support this too.

In Theia we already have some proxy preferences. It would make sense to reuse them for the Open AI API too.

![image](https://github.com/user-attachments/assets/abe2a30d-9c9f-4654-986f-49cd1529b370)


---

Note that your screenshot shows your whole Open AI API key. In case this was on accident I would recommend deactivating this key.

**Reply by sdirix** [reply]
I opened an issue for this: https://github.com/eclipse-theia/theia/issues/14324

**Reply by wss29** [reply]
Hi @sdirix 
   Thanks for your reply. Currently, I do not have a dedicated proxy URL for Open AI. My proxy is started on my computer for accessing some websites like Google.com by VPN. 
   So it seem that I can not use the ai feature when use open ai model until [#14324](https://github.com/eclipse-theia/theia/issues/14324) sovled
